lib/person.rb

```diff
@@ -0,0 +1,30 @@
+# lib/person.rb
+require_relative '../interfaces/nameable'
+
+class Person < Nameable
+  attr_reader :id, :name, :age, :rentals
+  attr_accessor :parent_permission
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def correct_name
+    @name
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
```

Comment:
- [ ] Please create a method for adding `rentals` in this class and remember the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
```ruby
 def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

Reply:
Thanks @DeliceLydia for noting the above.